api/src/main/java/org/apache/iceberg/variants/VariantValue.java

```diff
@@ -59,4 +59,21 @@ default VariantObject asObject() {
   default VariantArray asArray() {
     throw new IllegalArgumentException("Not an array: " + this);
   }
+
+  static VariantValue from(VariantMetadata metadata, ByteBuffer value) {
```

Comment:
Factory methods are copied into better places in the API now, rather than all being in `Variants`.